gateway/reverse_proxy.go

```diff
@@ -923,6 +923,17 @@ func (p *ReverseProxy) handleOutboundRequest(roundTripper *TykRoundTripper, outr
 		latency = time.Since(begin)
 	}()
 
+	if p.TykAPISpec.HasMock {
+		route, _, _ := p.TykAPISpec.OASRouter.FindRoute(outreq)
+		if route != nil {
```

Comment:
I suggest to move all these checks into mockResponse (without nesting ifs), add the validation on top of that function and return nil errors if appropriate/expected. Then we can unit test some things.

Reply:
Done